app/scripts/components/common/blocks/scrollytelling/index.tsx

```diff
@@ -402,12 +402,14 @@ function Scrollytelling(props) {
               }}
               classNames='reveal'
             >
-              <LayerLegend
-                id={`base-${activeChapterLayer.layer.id}`}
-                description={activeChapterLayer.layer.description}
-                title={activeChapterLayer.layer.name}
-                {...activeChapterLayer.layer.legend}
+            <LayerLegendContainer key={activeChapterLayer.layer.name}>
```

Comment:
@danielfdsilva  Legend does not update inside of the scrollymap and I am not sure why. Do you have a better insight?

Reply:
@hanbyul-here Added a commit to fix the animation. This was happening because the legend animates in and out with the help of `CSSTransition` which will wait for the transition to finish.
```
<CSSTransition
addEndListener={(node, done) => {
  if (!activeChapterLayer) return;
  node?.addEventListener('transitionend', done, false);
}}
>
```

When the wrapper `LayerLegendContainer` was added, the animation classes `reveal-*` were not moved to it form `LayerLegend`, causing the transition to never happen, leading to the `transitionend` event to never fire. Since the transition never finished the legend was not updated.

Reply:
Thank you for handling!